lib/main.js

```diff
@@ -87,6 +91,11 @@ export default {
           args.push("-xc++");
           args.push(...atom.config.get("linter-clang.clangDefaultCppFlags").split(/\s+/));
         }
+        if(activeEditor.getGrammar().name === "C++14") {
+          //const language = "c++";
+          args.push("-xc++");
+          args.push(...atom.config.get("linter-clang.clangDefaultCpp14Flags").split(/\s+/));
```

Comment:
I don't know if this method works with the settings page :-)

And even if it does, I think this change should be done in a dedicated PR ;-)